### PR TITLE
Add a git version guard for init.defaultBranch

### PR DESCRIPTION
**Pull request by mame**
Some specs use `git config --global init.defaultBranch main`, but the
configuration has been implemented since git 2.28.0.
I'm using git 2.25.1 (which is bundled in Ubuntu 20.04), so these specs
fail as follows.

```
Failures:

  1) bundle install git sources displays the correct default branch
...
       Diff:
       @@ -1,10 +1,19 @@
       -Using foo 1.0 from file:///home/mame/work/rubygems/bundler/tmp/1/libs/foo (at main@cdca7ca)
       ...
       +Using foo 1.0 from file:///home/mame/work/rubygems/bundler/tmp/1/libs/foo (at master@cdca7ca)
       ...
```

This changeset adds `:git => ">= 2.28.0"` to suppress the failures.

### CODE REVIEW

**Review by simi — APPROVED** [review]

File: bundler/spec/commands/newgem_spec.rb [hunk 1]
```diff
@@ -1149,15 +1149,15 @@ def create_temporary_dir(dir)
       it_behaves_like "--no-mit flag"
     end
 
-    context "with coc option in bundle config settings set to true" do
+    context "with coc option in bundle config settings set to true", :git => ">= 2.28.0" do
```
Comment (deivid-rodriguez) [inline bundler/spec/commands/newgem_spec.rb:1152]: Can you instead tag the specific spec that runs `git config --global init.defaultBranch main` and then checks that links to `main`are present in the generated README? It t would make it more apparent why the spec only works on some git versions, and allows you to run all the other specs that do pass.

Comment (mame) [reply bundler/spec/commands/newgem_spec.rb]: Is this right? https://github.com/rubygems/rubygems/pull/4533/commits/691b61751e7324564394cdab4ba6398170d50080

Comment (mame) [reply bundler/spec/commands/newgem_spec.rb]: Ah, I missed "then checks that links to mainare present in the generated README".

Comment (deivid-rodriguez) [reply bundler/spec/commands/newgem_spec.rb]: which I believe is the only spec that should be failing for you.

Comment (mame) [reply bundler/spec/commands/newgem_spec.rb]: Ahh, I see! https://github.com/rubygems/rubygems/pull/4533/commits/dab3fc5de156b47bd23dea2adfa5178c133a4c23 It works great on my machine.

Comment (deivid-rodriguez) [reply bundler/spec/commands/newgem_spec.rb]: Yes, I meant exactly that, except that you missed the ">=" in the filter! Thanks @mame!

Comment (mame) [reply bundler/spec/commands/newgem_spec.rb]: Fixed 😓

**Review by deivid-rodriguez — APPROVED** [review]
Thanks for your patience @mame, and sorry for the bad explanation :sweat_smile:.

**Pull request merged**